pkg/devfile/adapters/common/utils.go

```diff
@@ -0,0 +1,22 @@
+package common
+
+import (
+	"github.com/golang/glog"
+
+	"github.com/openshift/odo/pkg/devfile/versions"
+	"github.com/openshift/odo/pkg/devfile/versions/common"
+)
+
+// GetSupportedComponents iterates through the components in the devfile and returns a list of odo supported components
+func GetSupportedComponents(data versions.DevfileData) []common.DevfileComponent {
```

Comment:
Could you add a unit test for GetSupportedComponents?

Reply:
makes sense, let me see